### PR TITLE
Customized course checklist

### PR DESCRIPTION
**Pull request by arthurian**
This PR modifies the course setup checklist per ATG's requirements. 

_Files:_
- **js/checklist.js** - intended to be added to the global JS include 
- **css/checklist.css** -- intended to be added to the global CSS include. 
- **img/john_harvard_statue.jpg** - referenced from the CSS; used for the background image
- **LICENSE** - BSD 3-clause 

_Source Repository:_

[Harvard-ATG/canvas-course-checklist](https://github.com/Harvard-ATG/canvas-course-checklist)

What do you think? @harvard-canvas-branding/owners

### CODE REVIEW

File: js/checklist.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+	//----- DEBUG -----
+	if(DEBUG) {
+		$.getJSON("/api/v1/accounts/39/external_tools").done(function(data) {
+			console.log("List of tools for account_id 39:");
+			$.each(data, function(idx, tool) { 
+				console.log("tool consumer key:", tool.consumer_key, "tool id:", tool.id);
+			});
+		});
+		console.log("customized setup checklist: ", ListItems);
```
Comment (roderickmorales) [inline js/checklist.js]: do you need this console.log?

File: js/checklist.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+			'</style>'
+		].join(''));
+	}
+
+	//----- DEBUG -----
+	if(DEBUG) {
+		$.getJSON("/api/v1/accounts/39/external_tools").done(function(data) {
+			console.log("List of tools for account_id 39:");
+			$.each(data, function(idx, tool) { 
+				console.log("tool consumer key:", tool.consumer_key, "tool id:", tool.id);
```
Comment (roderickmorales) [inline js/checklist.js]: Do you need this console.log?

File: js/checklist.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			'<style>',
+			'.ic-wizard-box {',
+				'background: url("'+BACKGROUND_IMG_URL+'") no-repeat center center !important;',
+				'background-size: 100% auto !important;',
+			'</style>'
+		].join(''));
+	}
+
+	//----- DEBUG -----
+	if(DEBUG) {
```
Comment (roderickmorales) [inline js/checklist.js:148]: Ignore my console.log comments. I just saw that it was part of this (DEBUG) block.

File: js/checklist.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	var html = "<p>If you would like to incorporate content from a previous iSite, please contact the Academic Technology Group at <a href=\"mailto:atg@fas.harvard.edu\">atg@fas.harvard.edu</a>.</p>";
+	var is_content_migration_page = /^\/courses\/\d+\/content_migrations/.test(window.location.pathname);
+	var poll_for_element = pollForElement("#migrationConverterContainer > h1", 20, 100, function($el) {
+		$el.after(html);
+	});
+
+	if (is_content_migration_page) {
+		poll_for_element();
+	}
+
```
Comment (roderickmorales) [inline js/checklist.js:26]: Can we add some comments here?

Comment (arthurian) [inline js/checklist.js:26]: Any line or lines in particular? The block comment at the top of the`require()` block attempts to describe what the code does and why it's doing it that way. Are you looking for more of a line-by-line description?

Comment (arthurian) [inline js/checklist.js:26]: I just realized you were referring to the `pollForElement()` function, I think. I'll add some comments to that.

File: js/checklist.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+		url: BASE_COURSE_URL + "/external_tools/" + POLICY_WIZARD_TOOL_ID,
+		iconClass: 'icon-educators'
+	});
+	
+	//----- CHANGE #5 -----
+	// Modify background image.
+	if (BACKGROUND_IMG_URL) {
+		$("head").append([
+			'<style>',
+			'.ic-wizard-box {',
```
Comment (cmurtaugh) [inline js/checklist.js]: If this CSS bit was supplied in a separate .css file, then the background image URL could just be a relative (to the CSS location) path, right? That'd make deployment easier/safer.  Are there drawbacks to that approach?

Comment (arthurian) [inline js/checklist.js]: You're absolutely right, that would be a much cleaner approach and make deployment easier. I'll go ahead and make that change and push it up.

**Comment by arthurian** [comment]
@cmurtaugh I added a separate CSS file for the background image. Anything else we should do to get this ready for deployment?

**Pull request merged**